packages/sdk38/src/lcdapi/staking.spec.ts

```diff
@@ -0,0 +1,508 @@
+          memo: memo,
+          signatures: [signature],
+        };
+
+        const receipt = await client.postTx(tx);
+        assert(!isPostTxFailure(receipt));
+      }
+
+      await sleep(75); // wait until transactions are indexed
+    }
+  });
+
+  describe("delegatorDelegations", () => {
+    it("works", async () => {
+      pendingWithoutWasmd();
+      const client = makeStakingClient(wasmd.endpoint);
+      const response = await client.staking.delegatorDelegations(faucet.address);
+      expect(response).toEqual({
+        height: jasmine.stringMatching(nonNegativeIntegerMatcher),
+        result: [
+          {
+            delegator_address: "cosmos1pkptre7fdkl6gfrzlesjjvhxhlc3r4gmmk8rs6",
+            validator_address: "cosmosvaloper1gjvanqxc774u6ed9thj4gpn9gj5zus5u32enqn",
+            shares: jasmine.stringMatching(bigDecimalMatcher),
+            balance: { denom: "ustake", amount: jasmine.stringMatching(nonNegativeIntegerMatcher) },
+          },
+        ],
+      });
+    });
+  });
+
+  describe("delegatorUnbondingDelegations", () => {
+    it("works", async () => {
+      pendingWithoutWasmd();
+      const client = makeStakingClient(wasmd.endpoint);
+      const response = await client.staking.delegatorUnbondingDelegations(faucet.address);
+      expect(response.height).toMatch(nonNegativeIntegerMatcher);
+      expect(response.result.length).toEqual(1);
+      expect(response.result[0].delegator_address).toEqual("cosmos1pkptre7fdkl6gfrzlesjjvhxhlc3r4gmmk8rs6");
+      expect(response.result[0].validator_address).toEqual(
+        "cosmosvaloper1gjvanqxc774u6ed9thj4gpn9gj5zus5u32enqn",
+      );
+      expect(response.result[0].entries.length).toBeGreaterThanOrEqual(1);
+      expect(response.result[0].entries[0]).toEqual({
+        creation_height: jasmine.stringMatching(nonNegativeIntegerMatcher),
+        completion_time: jasmine.stringMatching(dateTimeStampMatcher),
+        initial_balance: "100",
+        balance: "100",
+      });
+    });
+  });
+
+  describe("delegatorTransactions", () => {
+    it("works", async () => {
+      pendingWithoutWasmd();
+      const client = makeStakingClient(wasmd.endpoint);
+      const response = await client.staking.delegatorTransactions(faucet.address);
+      expect(response.length).toEqual(3);
+    });
+  });
+
+  describe("delegatorValidators", () => {
+    it("works", async () => {
+      pendingWithoutWasmd();
+      const client = makeStakingClient(wasmd.endpoint);
+      const response = await client.staking.delegatorValidators(faucet.address);
+      expect(response).toEqual({
+        height: jasmine.stringMatching(nonNegativeIntegerMatcher),
+        result: [
+          {
+            operator_address: "cosmosvaloper1gjvanqxc774u6ed9thj4gpn9gj5zus5u32enqn",
+            consensus_pubkey:
+              "cosmosvalconspub1zcjduepqau36ht2r742jh230pxlu4wjmwcmkwpeqava80acphsu87vt5xlpqx6g7qh",
+            jailed: false,
+            status: 2,
```

Comment:
Can we create an enum with exlicit raw values for those numeric status representations?

Reply:
Since this appears in the results when we filter for "bonded", I'm assuming the status should be
```
Unbonded: 0
Unbonding: 1
Bonded: 2
```
Even though in the Go code it's 1–3.

Reply:
0, 1, 2 is what I found last time as well: https://github.com/cosmos/cosmos-sdk/blob/v0.38.5/types/staking.go#L43-L49

Reply:
Ah ok then we may need to update this later. See current master: https://github.com/cosmos/cosmos-sdk/blob/53bf227/types/staking.go

Reply:
Oh, and we already have

```ts
/**
 * Numeric bonding status
 *
 * @see https://github.com/cosmos/cosmos-sdk/blob/v0.38.5/types/staking.go#L43-L49
 */
export enum BondStatus {
  Unbonded = 0,
  Unbonding = 1,
  Bonded = 2,
}
```

Reply:
Yeah this update is good since protobuf does not differentiate between zero and unset